packages/flutter_tools/test/general.shard/analytics_test.dart

```diff
@@ -32,46 +32,44 @@ void main() {
   });
 
   group('analytics', () {
-    Directory tempDir;
+    FileSystem fileSystem;
     MockFlutterConfig mockFlutterConfig;
 
     setUp(() {
       Cache.flutterRoot = '../..';
-      tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_analytics_test.');
+      fileSystem = MemoryFileSystem.test();
       mockFlutterConfig = MockFlutterConfig();
     });
 
-    tearDown(() {
-      tryToDelete(tempDir);
-    });
-
     // Ensure we don't send anything when analytics is disabled.
     testUsingContext("doesn't send when disabled", () async {
       int count = 0;
       globals.flutterUsage.onSend.listen((Map<String, dynamic> data) => count++);
 
+      final FlutterCommand command = FakeFlutterCommand();
+      final CommandRunner<void>runner = createTestCommandRunner(command);
+
       globals.flutterUsage.enabled = false;
-      await createProject(tempDir);
+      await runner.run(<String>['fake']);
       expect(count, 0);
 
       globals.flutterUsage.enabled = true;
-      await createProject(tempDir);
+      await runner.run(<String>['fake']);
       expect(count, globals.flutterUsage.isFirstRun ? 0 : 4);
```

Comment:
This seems odd...

Reply:
perhaps it needed the real project so it could run the real doctor?

Reply:
I mean the `globals.flutterUsage.isFirstRun` check part circa https://github.com/flutter/flutter/pull/3523/files#diff-1202b4fb926dd6b56d4fb16f50f6baedR50

Reply:
`globals.flutterUsage.isFirstRun` is always fails because it's backed by a `LogToFileAnalytics` which is a `AnalyticsMock` which hardcodes `isFirstRun` to false.